cas/vaultcas/vaultcas.go

```diff
@@ -0,0 +1,322 @@
+package vaultcas
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/cas/apiv1"
+
+	vault "github.com/hashicorp/vault/api"
+	auth "github.com/hashicorp/vault/api/auth/approle"
+	certutil "github.com/hashicorp/vault/sdk/helper/certutil"
+	mapstructure "github.com/mitchellh/mapstructure"
+)
+
+func init() {
+	apiv1.Register(apiv1.VaultCAS, func(ctx context.Context, opts apiv1.Options) (apiv1.CertificateAuthorityService, error) {
+		return New(ctx, opts)
+	})
+}
+
+type VaultOptions struct {
+	PKI             string `json:"pki,omitempty"`
+	PKIRole         string `json:"pkiRole,omitempty`
+	PKIRoleRSA      string `json:"pkiRoleRSA,omitempty`
+	PKIRoleEC       string `json:"pkiRoleEC,omitempty`
+	PKIRoleED25519  string `json:"PKIRoleED25519,omitempty`
```

Comment:
Fix missing double quotes and lower case `d`
```suggestion
	PKIRoleRSA      string `json:"pkiRoleRSA,omitempty"`
	PKIRoleEC       string `json:"pkiRoleEC,omitempty"`
	PKIRoleEd25519  string `json:"PKIRoleEd25519,omitempty"`
```

Reply:
This is not yet fixed.